### ISSUE TITLE
[Accessibility]: Notebooks read-only 

### ISSUE THREAD

**Issue by marisak** [sourcegraph/sourcegraph#35681]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/35142

### Problem description

For the read-only view of Notebooks, the following ARC tests were triggered: 
> aria-label is used on an element which does not allow this attribute 
and 
> invalid aria role specified 

Many of these tests are in the main navigation of Notebooks, which does not apply here. However, the document outline and the notebook blocks are affected by this issue. 

### Expected behavior

From the ARC toolkit: 
> Remove aria-label attribute, and use visually hidden text instead
and 
> use a valid aria role 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
